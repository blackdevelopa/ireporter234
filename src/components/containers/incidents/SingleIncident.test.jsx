import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import configMockStore from 'redux-mock-store';
import ConnectedSingleIntervention, {
  SingleIntervention,
} from './SingleIntervention';
import ConnectedSingleRedflag, { SingleRedFlag } from './SingleRedFlag';

const props = {
  interventionInfo: {},
  isLoading: true,
  match: {
    params: {
      id: 10,
    },
  },
  fetchSingleIncident: jest.fn(),
  singleIntervention: {},
  singleRedflag: {},
};

const initialState = {
  isLoading: false,
  success: false,
  error: false,
  incident: {
    singleIncident: {},
  },
  auth: {
    isAuthenticated: false,
  },
};

jest.mock('../../../redux/actions/auth/login');
jest.mock('../../../store/');
const mockStore = configMockStore([thunk]);
const store = mockStore(initialState);

const MyContext = React.createContext();

const connectedWrapper = mount(
  <Provider store={store} context={MyContext}>
    <Router>
      <div>
        <ConnectedSingleIntervention context={MyContext} {...props} />
        <ConnectedSingleRedflag context={MyContext} {...props} />
      </div>
    </Router>
  </Provider>
);

describe('<Login> ', () => {
  it('should render without login component', () => {
    expect(connectedWrapper).toMatchSnapshot();
  });

  it('should render loading when "isLoading is true" ', () => {
    const interventionWrapper = mount(<SingleIntervention {...props} />);
    const redflagWrapper = mount(<SingleRedFlag {...props} />);
    expect(interventionWrapper.find('.loading')).toHaveLength(1);
    expect(redflagWrapper.find('.loading')).toHaveLength(1);
  });

  it('should should render no incidents text', () => {
    const interventionWrapper = mount(<SingleIntervention {...props} />);
    const redflagWrapper = mount(<SingleRedFlag {...props} />);
    interventionWrapper.setProps({ isLoading: false });
    redflagWrapper.setProps({ isLoading: false });
    expect(redflagWrapper.find('.loading')).toHaveLength(0);
    expect(redflagWrapper.find('.no-incidents')).toHaveLength(1);
  });

  it('should should render  single intervention', () => {
    const wrapper = mount(<SingleIntervention {...props} />);
    wrapper.setProps({ isLoading: false });
    wrapper.setProps({
      singleIntervention: {
        id: 1,
        comment: 'the comment',
        location: 'the location',
        images: 'image.png',
      },
    });
    const description = wrapper.find('.description');
    const header = wrapper.find('.header');
    expect(wrapper.find('.loading')).toHaveLength(0);
    expect(wrapper.find('.no-incidents')).toHaveLength(0);
    expect(description.text().includes('the comment')).toBe(true);
    expect(header.text().includes('the location')).toBe(true);
  });
});
