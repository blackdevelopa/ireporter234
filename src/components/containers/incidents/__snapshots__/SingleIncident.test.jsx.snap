// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login />  should render without crashing 1`] = `
<Provider
  context={
    Object {
      "$$typeof": Symbol(react.context),
      "Consumer": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": [Circular],
      },
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_currentRenderer": Object {},
      "_currentRenderer2": null,
      "_currentValue": undefined,
      "_currentValue2": undefined,
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "lrplgp",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "lrplgp",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <Connect(withRouter(SingleIntervention))
          context={
            Object {
              "$$typeof": Symbol(react.context),
              "Consumer": Object {
                "$$typeof": Symbol(react.context),
                "_calculateChangedBits": null,
                "_context": [Circular],
              },
              "Provider": Object {
                "$$typeof": Symbol(react.provider),
                "_context": [Circular],
              },
              "_calculateChangedBits": null,
              "_currentRenderer": Object {},
              "_currentRenderer2": null,
              "_currentValue": undefined,
              "_currentValue2": undefined,
            }
          }
          fetchSingleIncident={[MockFunction]}
          interventionInfo={Object {}}
          isLoading={true}
          match={
            Object {
              "params": Object {
                "id": 10,
              },
            }
          }
          singleIntervention={Object {}}
          singleRedflag={Object {}}
        >
          <withRouter(SingleIntervention)
            context={
              Object {
                "$$typeof": Symbol(react.context),
                "Consumer": Object {
                  "$$typeof": Symbol(react.context),
                  "_calculateChangedBits": null,
                  "_context": [Circular],
                },
                "Provider": Object {
                  "$$typeof": Symbol(react.provider),
                  "_context": [Circular],
                },
                "_calculateChangedBits": null,
                "_currentRenderer": Object {},
                "_currentRenderer2": null,
                "_currentValue": undefined,
                "_currentValue2": undefined,
              }
            }
            fetchSingleIncident={[Function]}
            interventionInfo={Object {}}
            match={
              Object {
                "params": Object {
                  "id": 10,
                },
              }
            }
            singleIntervention={Object {}}
            singleRedflag={Object {}}
          >
            <Route>
              <SingleIntervention
                context={
                  Object {
                    "$$typeof": Symbol(react.context),
                    "Consumer": Object {
                      "$$typeof": Symbol(react.context),
                      "_calculateChangedBits": null,
                      "_context": [Circular],
                    },
                    "Provider": Object {
                      "$$typeof": Symbol(react.provider),
                      "_context": [Circular],
                    },
                    "_calculateChangedBits": null,
                    "_currentRenderer": Object {},
                    "_currentRenderer2": null,
                    "_currentValue": undefined,
                    "_currentValue2": undefined,
                  }
                }
                fetchSingleIncident={[Function]}
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "lrplgp",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "lrplgp",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                interventionInfo={Object {}}
                location={
                  Object {
                    "hash": "",
                    "key": "lrplgp",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                singleIntervention={Object {}}
                singleRedflag={Object {}}
              >
                <div>
                  <h1>
                    There is no incident here
                  </h1>
                </div>
              </SingleIntervention>
            </Route>
          </withRouter(SingleIntervention)>
        </Connect(withRouter(SingleIntervention))>
        <Connect(withRouter(SingleRedFlag))
          context={
            Object {
              "$$typeof": Symbol(react.context),
              "Consumer": Object {
                "$$typeof": Symbol(react.context),
                "_calculateChangedBits": null,
                "_context": [Circular],
              },
              "Provider": Object {
                "$$typeof": Symbol(react.provider),
                "_context": [Circular],
              },
              "_calculateChangedBits": null,
              "_currentRenderer": Object {},
              "_currentRenderer2": null,
              "_currentValue": undefined,
              "_currentValue2": undefined,
            }
          }
          fetchSingleIncident={[MockFunction]}
          interventionInfo={Object {}}
          isLoading={true}
          match={
            Object {
              "params": Object {
                "id": 10,
              },
            }
          }
          singleIntervention={Object {}}
          singleRedflag={Object {}}
        >
          <withRouter(SingleRedFlag)
            context={
              Object {
                "$$typeof": Symbol(react.context),
                "Consumer": Object {
                  "$$typeof": Symbol(react.context),
                  "_calculateChangedBits": null,
                  "_context": [Circular],
                },
                "Provider": Object {
                  "$$typeof": Symbol(react.provider),
                  "_context": [Circular],
                },
                "_calculateChangedBits": null,
                "_currentRenderer": Object {},
                "_currentRenderer2": null,
                "_currentValue": undefined,
                "_currentValue2": undefined,
              }
            }
            fetchSingleIncident={[Function]}
            interventionInfo={Object {}}
            isLoading={true}
            match={
              Object {
                "params": Object {
                  "id": 10,
                },
              }
            }
            singleIntervention={Object {}}
            singleRedflag={Object {}}
          >
            <Route>
              <SingleRedFlag
                context={
                  Object {
                    "$$typeof": Symbol(react.context),
                    "Consumer": Object {
                      "$$typeof": Symbol(react.context),
                      "_calculateChangedBits": null,
                      "_context": [Circular],
                    },
                    "Provider": Object {
                      "$$typeof": Symbol(react.provider),
                      "_context": [Circular],
                    },
                    "_calculateChangedBits": null,
                    "_currentRenderer": Object {},
                    "_currentRenderer2": null,
                    "_currentValue": undefined,
                    "_currentValue2": undefined,
                  }
                }
                fetchSingleIncident={[Function]}
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "lrplgp",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "lrplgp",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                interventionInfo={Object {}}
                isLoading={true}
                location={
                  Object {
                    "hash": "",
                    "key": "lrplgp",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                singleIntervention={Object {}}
                singleRedflag={Object {}}
              >
                <div
                  className="loading"
                >
                  Loading...
                </div>
              </SingleRedFlag>
            </Route>
          </withRouter(SingleRedFlag)>
        </Connect(withRouter(SingleRedFlag))>
      </div>
    </Router>
  </MemoryRouter>
</Provider>
`;
